genset_whisperpower_ddc.orogen

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+name "genset_whisperpower_ddc"
+
+import_types_from "std"
+import_types_from "base"
+
+using_library "genset_whisperpower_ddc"
+
+import_types_from "genset_whisperpower_ddc/ControlCommand.hpp"
+import_types_from "genset_whisperpower_ddc/GeneratorState.hpp"
+import_types_from "genset_whisperpower_ddc/RunTimeState.hpp"
+
+# If new data types need to be defined, they have to be put in a separate C++
+# header, and this header will be loaded here
+import_types_from "genset_whisperpower_ddcTypes.hpp"
+
+using_task_library "iodrivers_base"
+
+# Declare a new task context (i.e., a component)
+#
+# The corresponding C++ class can be edited in tasks/Task.hpp and
+# tasks/Task.cpp, and will be put in the genset_whisperpower_ddc namespace.
+task_context "Task", subclasses: "iodrivers_base::Task" do
+    # This is the default from now on, and should not be removed. Rock will
+    # transition to a setup where all components use a configuration step.
+    needs_configuration
+
+    # An input port, i.e. an object from which the component gets data from
+    # other components' outputs
+    #
+    # Data can be retrieved using _input.read(value), which returns true if data
+    # was available, and false otherwise. _input.connected() returns if this
+    # input is connected to an output or not.
+    input_port "control_cmd", "/genset_whisperpower_ddc/ControlCommand"
+
+    # An output port, i.e. an object to which the component pushes data so that
+    # it is transmitted to other components' inputs
+    #
+    # Data can be written using _output.write(value). _output.connected() returns
+    # if this output is connected to an input or not.
+    output_port "generator_state", "/genset_whisperpower_ddc/GeneratorState"
+
+    output_port "run_time_state", "/genset_whisperpower_ddc/RunTimeState"
+
+    # If you want that component's updateHook() to be executed when the "input"
+    # port gets data, uncomment this and comment the 'periodic' line
+    # port_driven "input"
+    # By default, the task will be periodic with a period of 0.1
+    periodic 0.1
```

Comment:
This component should actually be left to `fd_driven` by default (which is the default when you subclass `iodrivers_base::Task`)